### PR TITLE
Await get options http client

### PR DESCRIPTION
**Pull request by Ashniu123**

### CODE REVIEW

**Comment by Ashniu123** [comment]
Failing test case doesn't seem to be connected to my changes.

File: src/client/httpClient.js [hunk 1]
```diff
@@ -30,15 +30,15 @@ export default (
     });
   });
 
-  store.on('watch', (query, options) => {
+  store.on('watch', async (query, options) => {
```
Comment (aravindet) [inline src/client/httpClient.js:33]: This is actually wrong. The watch handler must not be an async function (aka function returning a promise), it must be an async generator function (`async function*`) or a function that returns an async iterable.

This is likely the reason for the test failure.

Comment (aravindet) [reply src/client/httpClient.js:33]: If you do this, replace the `return makeStream(...)` statements with `yield* makeStream(...)`. The alternative is to use a bunch of .then()'s and keep it a regular non-async function.

Comment (Ashniu123) [reply src/client/httpClient.js:33]: Yeah. My bad for jumping the gun. I'll fix this and also add a test case for watch.

Comment (Ashniu123) [reply src/client/httpClient.js]: Test case a bit complex (involves mocking EventSource). Will add it some other day.

**Pull request merged**